### PR TITLE
Cypress testing

### PR DESCRIPTION
**Pull request by novaraptur**
# Description

Added sad path testing / stubbed server responses. Added some additional UX / user flow tests.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Console Dev Tools
- [X] Cypress

### CODE REVIEW

File: cypress/integration/App/App.spec.js [hunk 1]
```diff
@@ -6,7 +6,50 @@ describe('App', () => {
...
+  it('Should allow the user to go back to the game start page', () => {
+    cy.get('button').contains('Saved Games').click();
+    cy.url().should('eq', 'http://localhost:3000/saved-games');
+    cy.get('h2').should('contain', 'Past Games');
+    cy.get('button').contains('Back').click();
+    cy.url().should('eq', 'http://localhost:3000/');
+    cy.get('label').contains('Number of Categories:')
+  });
 
+  it('Should not allow the user to go to the game page without selecting a number of categories', () => {
```
Comment (hoomberto) [inline cypress/integration/App/App.spec.js:26]: Great check implemented here!

File: cypress/integration/Game/Game.spec.js [hunk 2]
```diff
@@ -1,35 +1,26 @@
 describe('Game', () => {
   beforeEach(() => {
     cy.visit('http://localhost:3000');
-
-    cy.intercept('GET', 'https://better-jeopardy-api.herokuapp.com/api/v1/questions', {
+    cy.intercept('GET', 'https://better-jeopardy-api-v2.herokuapp.com/api/v1/questions', {
       statusCode: 200,
       fixture: 'allQuestions.json'
```
Comment (hoomberto) [inline cypress/integration/Game/Game.spec.js:6]: Thanks Shawn for setting up these fixtures!

**Review by hoomberto — APPROVED** [review]

File: cypress/integration/Game/Game.spec.js (hunk 2 above)
Comment (shawnmcmahon) [reply cypress/integration/Game/Game.spec.js:6]: You got it!

File: cypress/integration/App/App.spec.js [hunk 3]
```diff
@@ -50,5 +50,6 @@ describe('App', () => {
     cy.visit('http://localhost:3000');
     cy.get('button').contains('Start Game').click();
     cy.url().should('eq', 'http://localhost:3000/');
+    cy.get('.error-message').contains('Error: 500 Internal Server Error')
```
Comment (shawnmcmahon) [inline cypress/integration/App/App.spec.js:53]: Nice sad path testing!

**Pull request merged**